### DISCUSSION TITLE
Mapping of metadata.lock-version to Poetry version(s)

### DISCUSSION THREAD

**Discussion by rarkins** [python-poetry/poetry#7891] [Q&A] [answered]
Hi, is there any mapping existing of lock-version to Poetry version(s)? e.g.
- 1.0 -> ?
- 1.1 -> ?
- 2.0 -> ?

Apart from looking at this lock-version field, is there any alternative way to determining which version of Poetry was responsible for a particular pyproject.toml/poetry.lock combination?

**Comment by rarkins** [comment]
Using git blame, I think I located the commits which set these versions:

- 1.0 added in 1.0.10: https://github.com/python-poetry/poetry/commit/cd72aed852a61cb054ede4971eec048f17d98b14
- 1.1 added in 1.1.0: https://github.com/python-poetry/poetry/commit/3222fe3c6b43ceec91b59dd9ca4dc04241b5166a
- 2.0 added in 1.3.0: https://github.com/python-poetry/poetry/commit/c061ac55ac50498281eb35bf15cfe8a593d676ee

**Reply by radoering** [reply, marked as answer]
Poetry 1.4 and later write the poetry version used to create a lock file in the comment on the first line of the lock file. See e.g. https://github.com/python-poetry/poetry/blob/master/poetry.lock#L1
